### PR TITLE
fixCrashOnCompetitionDetails

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/1986
Resolves: https://github.com/daostack/alchemy/issues/1989

### CODE REVIEW

File: src/components/Plugin/ContributionRewardExtRewarders/Competition/Details.tsx [hunk 1]
```diff
@@ -276,7 +275,7 @@ class CompetitionDetails extends React.Component<IProps, IStateProps> {
     this.disqusConfig.identifier = `competition-${proposalState.id}`;
 
     return <React.Fragment>
-      <BreadcrumbsItem weight={1} to={`/dao/${daoState.address}/plugin/${proposalState.plugin.id}/crx`}>{pluginName(proposalState.plugin.entity.coreState, proposalState.plugin.entity.coreState.address)}</BreadcrumbsItem>
+      <BreadcrumbsItem weight={1} to={`/dao/${daoState.address}/plugin/${proposalState.plugin.id}/crx`}>Competition</BreadcrumbsItem>
```
Comment (orenyodfat) [inline src/components/Plugin/ContributionRewardExtRewarders/Competition/Details.tsx:278]: does pluginName function has an issue?

Comment (dkent600) [reply src/components/Plugin/ContributionRewardExtRewarders/Competition/Details.tsx:278]: good catch!  Yes there was something wrong with `pluginName`.

Note that having fixed that, it reveals the problem described in [this ticket](https://github.com/daostack/alchemy/issues/1985), which is a separate problem.

File: src/lib/pluginUtils.ts [hunk 2]
```diff
@@ -102,11 +102,8 @@ export function pluginName(plugin: IPluginState|IContractInfo, fallback?: string
     name = rewarderContractName(plugin as IContributionRewardExtState);
 
     if (!name) {
-      name = "ContributionRewardExt";
-    } else {
-      // add spaces before capital letters to approximate a human-readable title
-      name = `${plugin.name[0]}${plugin.name.slice(1).replace(/([A-Z])/g, " $1")}`;
-    }
+      name = PLUGIN_NAMES["ContributionRewardExt"];
```
Comment (orenyodfat) [inline src/lib/pluginUtils.ts:105]: so this is only for cases where there is no rewarder?

Comment (orenyodfat) [inline src/lib/pluginUtils.ts:105]: should it just be here PLUGIN_NAMES[plugin.name as keyof typeof PLUGIN_NAMES]; ?

File: src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts [hunk 3]
```diff
@@ -61,3 +61,21 @@ export const getCrxRewarderComponent = (pluginState: IContributionRewardExtState
...
+export interface IEntityState {
+  id: string;
+}
+
+export const getCrxRewarderProposalClass =
+  (pluginState: IContributionRewardExtState): new (context: Arc, idOrOpts: string | IEntityState) => any => {
+
+    const contractName = rewarderContractName(pluginState, false);
+
+    switch (contractName) {
```
Comment (orenyodfat) [inline src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts:77]: can this be `switch (rewarderContractName(pluginState, false)) {`

File: src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts [hunk 4]
```diff
@@ -61,3 +61,21 @@ export const getCrxRewarderComponent = (pluginState: IContributionRewardExtState
...
+      case "Competition":
+        return CompetitionProposal;
```
Comment (orenyodfat) [inline src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts:79]: what the function will return if the rewarder is not `Competition` ?

File: src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts (hunk 3 above)
Comment (dkent600) [reply src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts]: Done

File: src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts (hunk 4 above)
Comment (dkent600) [reply src/components/Plugin/ContributionRewardExtRewarders/rewardersProps.ts:83]: null, done

File: src/lib/pluginUtils.ts (hunk 2 above)
Comment (dkent600) [reply src/lib/pluginUtils.ts:105]: yes, changed to `name = plugin.name;`

Comment (dkent600) [reply src/lib/pluginUtils.ts:105]: changed to `name = plugin.name`

**Pull request merged**